Classes/DDFileLogger.m

```diff
@@ -263,15 +263,13 @@ - (NSString *)logsDirectory {
     // We could do this check once, during initalization, and not bother again.
     // But this way the code continues to work if the directory gets deleted while the code is running.
 
-    if (![[NSFileManager defaultManager] fileExistsAtPath:_logsDirectory]) {
-        NSError *err = nil;
-
-        if (![[NSFileManager defaultManager] createDirectoryAtPath:_logsDirectory
-                                       withIntermediateDirectories:YES
-                                                        attributes:nil
-                                                             error:&err]) {
-            NSLogError(@"DDFileLogManagerDefault: Error creating logsDirectory: %@", err);
-        }
+    NSError *err = nil;
+    BOOL success = [[NSFileManager defaultManager] createDirectoryAtPath:_logsDirectory
+                                             withIntermediateDirectories:YES
+                                                              attributes:nil
+                                                                   error:&err];
+    if (success == NO) {
+        NSLogError(@"DDFileLogManagerDefault: Error creating logsDirectory: %@", err);
```

Comment:
For this, see the comment above `-[NSFileManager fileExistsAtPath]`

Reply:
when withIntermediateDirectories:YES is specified, the FileManager does not return any errors if the directory already exists.

Doc for return value on that method :
> true if the directory was created, true if createIntermediates is set and the directory already exists, or false if an error occurred.

https://developer.apple.com/documentation/foundation/filemanager/1407884-createdirectory

Reply:
Ah, you're right. Thanks for the heads up... 🙂